cmd/kueuectl/app/list/list_pods.go

```diff
@@ -0,0 +1,237 @@
+
+	_ = cmd.MarkFlagRequired("for")
+
+	return cmd
+}
+
+// Complete takes the command arguments and infers any remaining options.
+func (o *PodOptions) Complete(clientGetter util.ClientGetter) error {
+	var err error
+
+	o.Limit, err = listRequestLimit()
+	if err != nil {
+		return err
+	}
+	o.Namespace, _, err = clientGetter.ToRawKubeConfigLoader().Namespace()
+	if err != nil {
+		return err
+	}
+
+	o.Clientset, err = clientGetter.K8sClientSet()
+	if err != nil {
+		return err
+	}
+
+	// parse --for flag
+	if o.UserSpecifiedForObject != "" {
+		mapper, err := clientGetter.ToRESTMapper()
+		if err != nil {
+			return err
+		}
+		var found bool
+		o.ForGVK, o.ForName, found, err = decodeResourceTypeName(mapper, o.UserSpecifiedForObject)
+		if err != nil {
+			return err
+		}
+		if !found {
+			return fmt.Errorf("invalid value '%s' used in --for flag; value must be in the format [TYPE[.API-GROUP]/]NAME", o.UserSpecifiedForObject)
+		}
+	}
+
+	return nil
+}
+
+func (o *PodOptions) ToPrinter(headers bool) (printers.ResourcePrinterFunc, error) {
+	if !o.PrintFlags.OutputFlagSpecified() {
+		printer := newPodTablePrinter().
+			WithNamespace(o.AllNamespaces).
+			WithNoHeaders(headers)
+		return printer.PrintObj, nil
+	}
+
+	printer, err := o.PrintFlags.ToPrinter()
+	if err != nil {
+		return nil, err
+	}
+
+	return printer.PrintObj, nil
+}
+
+// Run prints the pods for a specific Job
+func (o *PodOptions) Run(ctx context.Context) error {
+	err := o.listPods(ctx)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// listPods lists the pods based on the given --for object
+func (o *PodOptions) listPods(ctx context.Context) error {
+	var totalCount int
+
+	namespace := o.Namespace
+	if o.AllNamespaces {
+		namespace = ""
+	}
+
+	opts := metav1.ListOptions{
+		LabelSelector: o.LabelSelector,
+		FieldSelector: o.FieldSelector,
+		Limit:         o.Limit,
+	}
+
+	tabWriter := printers.GetNewTabWriter(o.Out)
+	for {
+		podList, err := o.Clientset.CoreV1().Pods(namespace).List(ctx, opts)
+		if err != nil {
+			return err
+		}
+
+		podList.Items = o.filterPodsByOwnerRef(podList)
+
+		totalCount += len(podList.Items)
+		headers := totalCount == 0
+
+		printer, err := o.ToPrinter(headers)
+		if err != nil {
+			return err
+		}
+
+		if err = printer.PrintObj(podList, tabWriter); err != nil {
+			return err
+		}
+
+		if podList.Continue != "" {
+			opts.Continue = podList.Continue
+			continue
+		}
+
+		// handle if no filtered podList found
+		if totalCount == 0 {
+			if !o.AllNamespaces {
+				fmt.Fprintf(o.ErrOut, "No resources found in %s namespace.\n", o.Namespace)
+			} else {
+				fmt.Fprintln(o.ErrOut, "No resources found.")
+			}
+			return nil
+		}
+
+		if err = tabWriter.Flush(); err != nil {
+			return err
+		}
+
+		return nil
+	}
+}
+
+// filterPodsByOwnerRef filter pods based on the ownerReferences from the --for flag
+func (o *PodOptions) filterPodsByOwnerRef(podList *corev1.PodList) []corev1.Pod {
```

Comment:
The main issue with this is that it's checking only the first level of the ownership tree.

I think is is fine for batch/Jobs but for JobSet for example it's not sufficient (the pods of a JobSet are owned by Jobs which are owned by that JobSet), this is very likely the case for other jobs (RayJob, RayCluster and more).

Reply:
Thanks for catching that! Looks like we need to check ownerRef for each allowed integrations according to their Kinds?